server/api/runtime_handlers/base.py

```diff
@@ -1671,12 +1690,16 @@ def _ensure_run_state(
                     run_state=run_state,
                 )
 
+            elif run_state == RunStates.error:
+                # Try resolving the error reason
+                reason, message = self._resolve_container_error_status(runtime_resource)
+
         logger.info("Updating run state", run_uid=uid, run_state=run_state)
         run_updates = {
             "status.state": run_state,
             "status.last_update": now_date().isoformat(),
-            # run is not in terminal state, so reset reason and error
-            "status.reason": "",
+            "status.reason": reason or "",
+            "status.status_text": message or "",
```

Comment:
you don't need the `or ""`, they're already initialized as `""` and the resolve method can also return `""` and not None